master/internal/task/allocation.go

```diff
@@ -626,10 +626,14 @@ func (a *allocation) resourcesAllocated(msg *sproto.ResourcesAllocated) error {
 	}
 
 	now := time.Now().UTC()
+	taskStatStartTime := msg.JobSubmissionTime
+	if msg.JobSubmissionTime.IsZero() && a.req.Restore {
```

Comment:
even if we had this, in case of restores we'd want to use the request time? which would be the master start time when we're going through the restore path